[v1.5rc][F12-B4] HuatAh!

jonleeyz opened the pull request:
Auto updating Pull Request from our team repo to the course repo.

@melvintzw
@Der-Erlkonig
@zhangriqi

OuyangDanwen reviewed (commented):
@jonleeyz @zhangriqi @Der-Erlkonig @melvintzw Please change the product name in readme. It should not be AB4 at this point and make the PR before the tutorial. Thanks!

OuyangDanwen commented on docs/DeveloperGuide.adoc:
```diff
@@ -798,17 +802,65 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+|`* *` |user |view a route encompassing a given set of customers' locations |access the cost, time and effort involved in collecting from a set of customers
 
-|`*` |user with many persons in the address book |sort persons by name |locate a person easily
+|`*` |user |export my current view to another format |view specific information in another format or print out a hardcopy and pass the information to subordinates
```
Update MSS accordingly. I believe you do have new commands which have different workflow.

OuyangDanwen reviewed (commented):
A few things: 1. Please update the AboutUs, ContactUs pages asap. 2. You should re-write the whole UG to reflect the changes you intent to make for your own app. (the features should be up to V2.0) Mark them accordingly with the version. 3. More details should be documented in the DG as you go along. You might not have time to do it in one shot later. You should have a product survey for DG as well. Please take note of the above points. 4. You should update the PR title each week. Replace the "X" with correct version number. Due to the incomplete UG, I have to change the team component to 0 for last tutorial as the changes are required. (the lead tutor insisted for all teams to do this) @jonleeyz @melvintzw @Der-Erlkonig @zhangriqi

OuyangDanwen reviewed (commented):
For DG, it is good that you have included some code snippets, but do not overdo it. Use of UML diagrams is lacking. Please use your UML knowledge to explain your design. It is important to have diagrams in your project portfolio. UG should be in sync with you changes. Google map feature should be also included. Same applies to other new features. You can mark "Since V1.2" to indicate them clearly, similar to the marking "Coming in V2.0". Please mark sure your UG is ready before peer testing. @jonleeyz @melvintzw @Der-Erlkonig @zhangriqi

jonleeyz commented:
Hey @OuyangDanwen! Would there be a review of this PR for Milestone 1.3?

Otherwise, we will rename it soon to v1.4!

Thanks!

jonleeyz committed:
[UI.png] Update screenshot.

melvintzw committed:
Merge pull request #152 from jonleeyz/rebrand

Rebrand: LoanShark Manager -> HuatAh!

jonleeyz committed:
Remove ListCommand alias.

eencogneeto committed:
[HtmlWriter.java] improve look of PersonPage for Runner

jonleeyz committed:
Change LocateCommand alias.

jonleeyz committed:
Add home ResultDisplay message.

jonleeyz committed:
Change FindCommmand and LocateCommand templates.

melvintzw committed:
update user guide

melvintzw committed:
Improved readability of edit command success message

melvintzw committed:
Fix bug where "find w" does not work - where w is an alphanum character

melvintzw committed:
fix bugs where invalid values are provided for interest rate and money borrowed

jonleeyz committed:
[README.adoc] Update.

jonleeyz committed:
Update comments for Event classes.

eencogneeto committed:
[HtmlWriter.java] additional phone, email, address fields

jonleeyz committed:
Merge pull request #155 from melvintzw/VariousBugFixes

Various bug fixes

jonleeyz committed:
[AddressBookParserTest.java] Remove JUnit test for ListCommand alias.

eencogneeto committed:
[HtmlWriter.java] modify view for Customer

melvintzw committed:
Modify EditCommandParser and EditCommand to enable editing of Customer-only fields. Related exceptions are handled as well.

melvintzw committed:
Merge pull request #157 from jonleeyz/templateAndAliases-tweaks

Template and aliases tweaks

melvintzw committed:
Update UserGuide with edit command enhancement to accept customer-only fields

melvintzw committed:
Improve Locate Command messages

eencogneeto committed:
[HtmlWriter.java] Modify test

eencogneeto committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F12-B4/main into person_page_4

eencogneeto committed:
Merge pull request #159 from melvintzw/LocateCommandBugFix

Improve Locate Command messages

eencogneeto committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F12-B4/main into person_page_4

melvintzw committed:
Merge pull request #160 from Der-Erlkonig/person_page_4

UI BrowserPanel PersonPage improvement

melvintzw committed:
Ensure that BrowserPanel reflects most up-to-date Person information

Create FieldsChangedEvent
Subscribe BrowserPanel to FieldsChangedEvent
Post FieldsChangedEvent in EditCommand
Post FieldsChangedEvent in AssignCommand

melvintzw committed:
Remove temporarily the assert function for testing the selection of person card after Edit Command is executed.

jonleeyz committed:
Merge pull request #161 from melvintzw/EditCommand2

Edit command2

melvintzw committed:
Update EditCommand messages

melvintzw committed:
update UG for EditCommand

jonleeyz committed:
[UserGuide.adoc] Update LocateCommand entry.

melvintzw committed:
Improved Email invalid regex message to be slightly more concise and readable

melvintzw committed:
Update UG: known issues

melvintzw committed:
Merge pull request #163 from jonleeyz/UG-locate

[UserGuide.adoc] Update LocateCommand entry.

Commit:
Update Portfolio of zhangriqi

Commit:
fix typos

Commit:
mention ui

Commit:
project management

jonleeyz committed:
Merge pull request #164 from melvintzw/Msgs&UG&DG

Small updates to Messages, UG and DG

Commit:
fix trailing space

Commit:
fix travis error

eencogneeto committed:
[HtmlWriter.java] add amount borrowed field for customer to show

jonleeyz committed:
Merge branch 'master' into UpdatePortfolio

eencogneeto committed:
Merge branch 'master' into person_page_5

eencogneeto committed:
Merge pull request #166 from Der-Erlkonig/person_page_5

Add amount borrowed field for customer to show on UI

jonleeyz committed:
Merge branch 'master' into UpdatePortfolio

melvintzw committed:
Update Project Portfolio Page

melvintzw committed:
Merge branch 'master' into PPP

melvintzw committed:
Merge pull request #165 from zhangriqi/UpdatePortfolio

Update zhangriqi portfolio

melvintzw committed:
Merge branch 'master' into PPP

eencogneeto committed:
Update Project, DevGuide

eencogneeto committed:
Merge pull request #167 from melvintzw/PPP

Update Project Portfolio Page

eencogneeto committed:
Merge branch 'master' into PPP_1

jonleeyz committed:
Add include:: tags.

melvintzw committed:
Merge pull request #169 from Der-Erlkonig/PPP_1

Updates to Project, DevGuide

jonleeyz committed:
[jonleeyz.adoc] Update Project Portfolio Page.

jonleeyz committed:
[jonleeyz.adoc] Fix checkstyle violation: no EOF.

zhangriqi committed:
Merge pull request #170 from jonleeyz/PPP-jonlee

Update Project Portfolio Page

melvintzw committed:
Collate

zhangriqi committed:
Merge pull request #171 from melvintzw/master

Collate

jonleeyz committed:
[UserGuide.adoc] Add inline anchors and cross-references.

jonleeyz committed:
[UserGuide.adoc] Underline "Usage examples" header.

jonleeyz committed:
[UserGuide.adoc] Add syntax subsections.

jonleeyz committed:
[UserGuide.adoc] Add sections for remaining commands.

jonleeyz committed:
Update PPP introduction messages.

jonleeyz committed:
[UserGuide.adoc] Remove underlines from subsection headers.

jonleeyz committed:
[UserGuide.adoc] Fix inconsistencies.

jonleeyz committed:
Fix command message inconsistencies.

melvintzw committed:
edit Value Proposition, feature contribution and `delete` test case

jonleeyz committed:
Change AddCommand alias.

jonleeyz committed:
Change AssignCommand alias.

melvintzw committed:
Merge pull request #172 from jonleeyz/documentation-update

Update: Documentation

jonleeyz committed:
[ClearCommandTest.java] Add test: verifyGetCommandWordWorksCorrectly().

jonleeyz committed:
[ListCommandTest.java] Add test: verifyGetCommandWordWorksCorrectly().

jonleeyz committed:
[RedoCommandTest.java] Add test: verifyGetCommandWordWorksCorrectly().

jonleeyz committed:
[UndoCommandTest.java] Add test: verifyGetCommandWordWorksCorrectly().

melvintzw committed:
Improved various portions in the UserGuide

jonleeyz committed:
[ParserUtilTest.java] Update test implementation: parseAddress_invalidValue_throwsIllegalValueException().

jonleeyz committed:
[MoneyBorrowed.java] Rework error messages.

jonleeyz committed:
[ParserUtil.java] Update implementation: parseMoneyBorrowed().

melvintzw committed:
Add some regex tests for XmlAdaptedPerson.toModelType()
Prepared some files for JUnit tests

melvintzw committed:
in command summary, shifted redo section to after undo section

jonleeyz committed:
[LateInterest.java] Update implementation of LateInterest class.

jonleeyz committed:
[StandardInterest.java] Rework messages.

jonleeyz committed:
[MoneyBorrowed.java] Rework messages.

jonleeyz committed:
[ParserUtil.java] Update implementation: parseLateInterest().

jonleeyz committed:
[ParserUtil.java] Update implementation: parseMoneyBorrowed().

jonleeyz committed:
[ParserUtil.java] Update implementation: parseStandardInterest().

jonleeyz committed:
[ParserUtilTest.java] Add tests for parseMoneyBorrowed().

jonleeyz committed:
[ParserUtilTest.java] Add tests for parseStandardInterest().

jonleeyz committed:
[ParserUtilTest.java] Add tests for parseLateInterest().

jonleeyz committed:
Merge pull request #174 from melvintzw/Doc_v1.5

Doc v1.5

jonleeyz committed:
[ParserUtil.java] Tweak implementation: parseAddress().

jonleeyz committed:
[ParserUtilTest.java] Remove dysfunctional test.

jonleeyz committed:
[ParserUtilTest.java] Fix various checkstyle violations.

jonleeyz committed:
[ParserUtilTest.java] Fix various checkstyle violations.

jonleeyz committed:
Merge pull request #175 from jonleeyz/Tests-ParserAndCommands

JUnit Tests: Parser and Commands

jonleeyz committed:
Refactor: PREFIX_MONEYOWED -> PREFIX_MONEY_BORROWED.

jonleeyz committed:
[CommandTestUtil.java] Add new constants.

jonleeyz committed:
[EditPersonDescriptorBuilder.java] Add test helper methods.

jonleeyz committed:
[EditCommandParserTest.java] Update test implementations.

Test editing of MoneyBorrowed, OweStartDate,
OweDueDate and StandardInterest fields.

jonleeyz committed:
[ParserUtil.java] Update test method comments.

jonleeyz committed:
[ParserUtilTest.java] Add collate annotation.

jonleeyz committed:
[ParserUtil.java] Update test method signatures.

melvintzw committed:
change first two Customers to Runners.

melvintzw committed:
implement buildRunner()
change type of customers and runner

melvintzw committed:
Create JUnit test for assignCommand - unit test seems to work successfully

jonleeyz committed:
[AddressBookSystemTest.java] Move declaration and initialisation of GuiRobot.

From subclasses to the AddressBookSystemTest superclass.

jonleeyz committed:
[AddressBookSystemTest.java] Add new rule: EventsCollectorRule.

jonleeyz committed:
[AddCommandSystemTest.java] Reorganise tests.

jonleeyz committed:
[DeleteCommandSystemTest.java] Reorganise tests.

jonleeyz committed:
[EditCommandSystemTest.java] Reorganise tests.

jonleeyz committed:
[FindCommandSystemTest.java] Reorganise tests.

jonleeyz committed:
[SelectCommandSystemTest.java] Reorganise tests.

jonleeyz committed:
Add check for raising of PopulatePrefixesRequestEvent.

jonleeyz committed:
[HistoryCommandTest.java] Add test: verifyGetCommandWordWorksCorrectly().

jonleeyz committed:
[HelpCommandSystemTest.java] Restructure tests.

melvintzw committed:
Make an error message a psf String

melvintzw committed:
Implement two more working test cases (1 valid test case and 1 invalid test case)

melvintzw committed:
change error message to Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX

melvintzw committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F12-B4/main into JUnitTests

melvintzw committed:
change message references

jonleeyz committed:
[EditCommandSystemTest.java] Add comments.

jonleeyz committed:
Fix various checkstyle violations.

melvintzw committed:
Merge pull request #177 from jonleeyz/Tests-ParserAndCommands

Tests parser and commands

melvintzw committed:
change Runners to the last two persons in the list

melvintzw committed:
change runner index to the to refer to the correct runners in TypicalPersons.java

melvintzw committed:
change build type

melvintzw committed:
Merge branch 'master' into JUnitTests

jonleeyz committed:
Merge pull request #176 from melvintzw/JUnitTests

Some JUnit tests for AssignCommand

Commit:
Add more JUnit test for Locate Command

Commit:
Update ppp

jonleeyz committed:
[CommandBox.java] Implement helper merthods.

moveToPreviousPrefix(), moveToNextPrefix().

jonleeyz committed:
[CommandBox.java] Implement moveToPreviousPrefix().

jonleeyz committed:
[CommandBox.java] Implement moveToNextPrefix().

jonleeyz committed:
[CommandBox.java] Implement clearCurrentFieldOrPrefix().

jonleeyz committed:
[CommandBox.java] Add method calls.

jonleeyz committed:
[*Command.java] Made COMAND_TEMPLATE a constant.

jonleeyz committed:
[*CommandSystemTest.java] Update tests to reference constants.

jonleeyz committed:
[EditCommand.java] Update MESSAGE_USAGE and getCaretIndex().

Commit:
Change app name to HuatAh!

Commit:
fix failed test

melvintzw committed:
Merge pull request #180 from zhangriqi/UpdateAppName

Change app name to HuatAh!

Commit:
Return error message when locating a person without address.

jonleeyz committed:
[CommandBoxHandle.java] Implement helper methods.

setInput(), getCaretPosition() and setCaretPosition().

jonleeyz committed:
[CommandBoxTest.java] Add new tests.

Commit:
locate command JUnit test

melvintzw committed:
Merge branch 'master' into FixLocateCommandBug

jonleeyz committed:
[CommandBox.java] Fix checkstyle violations.

melvintzw committed:
Merge pull request #181 from zhangriqi/FixLocateCommandBug

Fix bug: Return error message when locating a person without address.

Commit:
Add more tests for LocateCommand

zhangriqi committed:
Merge pull request #183 from jonleeyz/commandbox-enhancements

CommandBox enhancements

eencogneeto committed:
add alias to shortcuts table
convert date to simple format in Customer's details

Commit:
Merge branch 'master' of git://github.com/CS2103JAN2018-F12-B4/main into LocateJUnitTest

Commit:
clean code

Commit:
Update project portfolio

melvintzw committed:
Merge pull request #179 from zhangriqi/LocateJUnitTest

Add more Locate Junit test

eencogneeto committed:
add Customer field oweStartDate

eencogneeto committed:
add customer field interest rate

eencogneeto committed:
change keyboard shortcut of exit to Alt+Q

eencogneeto committed:
edit colours of application (light blue)

eencogneeto committed:
swap email and address lines

eencogneeto committed:
modify test accordingly

melvintzw committed:
Merge branch 'master' into 14April_Afternoon

melvintzw committed:
Merge pull request #184 from Der-Erlkonig/14April_Afternoon

Colour and misc

melvintzw committed:
hardcoded setStyle() in PersonCard to detemine personCard colour based on personType

eencogneeto committed:
Merge pull request #185 from melvintzw/newUI

hardcoded setStyle() in PersonCard to detemine personCard colour base…

melvintzw committed:
change UI screenshot
change resultDisplay font color to black

melvintzw committed:
Implement JUnit tests for AssignCommandParser

melvintzw committed:
Implementing AssignCommandSystemTest

melvintzw committed:
Modify AssignCommandSystemTests so that it works (there seems to be a bug preventing the commented out tests from working - manual testing seems to work fine, there might be an issue with the UI updating slowly or asynchronously)

zhangriqi committed:
Merge pull request #186 from melvintzw/JUnitTests

JUnit tests for AssignCommand

jonleeyz committed:
[CommandBoxTest.java] Reimplement tests.

jonleeyz committed:
[LocateCommandParserTest.java] Add tests.

jonleeyz committed:
[LocateCommandParserTest.java] Disable 1 test: dysfunctional.

jonleeyz committed:
Fix checkstyle violations.

jonleeyz committed:
Merge branch 'master' into newUI

jonleeyz committed:
Merge pull request #187 from melvintzw/newUI

minor UI changes

melvintzw committed:
change list-cell colour to -main-colour

melvintzw committed:
fix unwanted lines in personListPanel

jonleeyz committed:
[UniquePersonListTest.java] Add new test: testHashCode_symmetric().

jonleeyz committed:
[AddressTest.java] Add new test: testHashCode_symmetric().

jonleeyz committed:
[EmailTest.java] Add new test: testHashCode_symmetric().

jonleeyz committed:
[NameTest.java] Add new test: testHashCode_symmetric().

jonleeyz committed:
[PhoneTest.java] Add new test: testHashCode_symmetric().

jonleeyz committed:
Merge pull request #189 from melvintzw/newUI

New ui

melvintzw committed:
Add diagrams for FindCommand and Model2

jonleeyz committed:
[AddressBookTest.java] Add new test: testHashCode_symmetric().

jonleeyz committed:
[AddressBookTest.java] Add new test: removeNonExistentPerson_throwsPersonNotFoundException().

Commit:
Add feature in v2.0

jonleeyz committed:
[LateInterestTest.java] Add new tests.

testToString_success(), testHashcode_symmetric().

jonleeyz committed:
[MoneyBorrowedTest.java] Add new tests.

testToString_success(), testHashcode_symmetric().

jonleeyz committed:
[StandardInterestTest.java] Add new tests.

testToString_success(), testHashcode_symmetric().

jonleeyz committed:
[UserPrefsTest.java] Add new tests.

testGetAddressBookName_Success(), testEquals_Symmetric(), testHashcode_Symmetric().

jonleeyz committed:
[SampleDataUtil.java] Remove unused method: defaultDate().

melvintzw committed:
Add more AssignCommand JUnit tests

melvintzw committed:
Merge pull request #191 from zhangriqi/DocumentationForv2.0

Add feature in v2.0

melvintzw committed:
Merge branch 'master' into JUnitTests

jonleeyz committed:
[AddressTest.java] Fix test: testHashcode_symmetric().

jonleeyz committed:
[PersonListPanel.java] Add whitespaces.

jonleeyz committed:
Fix checkstyle violation: rename package Customer to customer.

jonleeyz committed:
[UniquePersonListTest.java] Fix checkstyle violation: remove imports.

jonleeyz committed:
[UserPrefsTest.java] Fix checkstyle violation: rename methods.

jonleeyz committed:
[SampleDataUtil.java] Fix checkstyle violation: remove import.

jonleeyz committed:
Merge pull request #192 from melvintzw/JUnitTests

Add more AssignCommand JUnit tests

melvintzw committed:
update various documentation's app version to 1.5

melvintzw committed:
Merge pull request #193 from jonleeyz/JUnitTests

More JUnit Tests

eencogneeto committed:
Add tests and getter methods to facilitate tests

eencogneeto committed:
Update DeveloperGuide

jonleeyz committed:
[UI-start.PNG] Add new screenshot.

eencogneeto committed:
update DevGuide

melvintzw committed:
update melvintzw PPP

jonleeyz committed:
[UserGuide.adoc] Update numerous sections.

eencogneeto committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F12-B4/main into 15April

jonleeyz committed:
[UserGuide.adoc] Add section: Navigation Shortcuts.

jonleeyz committed:
Update version to v1.5.

jonleeyz committed:
[UI_old.jpg] Remove redundant image.

jonleeyz committed:
[README.adoc] Update.

melvintzw committed:
add tags(FindCommandEnhancement, ModelTwo) to DeveloperGuide
add tag references to FindCommandEnhancement, ModelTwo in PPP

eencogneeto committed:
Correct Test checkstyles, update DevGuide

melvintzw committed:
Merge pull request #195 from jonleeyz/documentation

Update documentation

eencogneeto committed:
Fix checkstyle

Commit:
Merge branch 'master' of git://github.com/CS2103JAN2018-F12-B4/main into PPP

melvintzw committed:
Merge branch 'master' into 15April

eencogneeto committed:
Merge pull request #194 from Der-Erlkonig/15April

HtmlWriter Tests, DevGuide

Commit:
Update documentation for locate

Commit:
Update documentation for command alias

Commit:
update ppp

jonleeyz committed:
[jonleeyz.adoc] Update.

jonleeyz committed:
[UserGuide.adoc] Update.

melvintzw committed:
update devGuide implementation section with Model 2.0

Commit:
Update developer guide

eencogneeto committed:
PPP and DevGuide

eencogneeto committed:
PPP

jonleeyz committed:
Merge pull request #197 from melvintzw/Doc_v1.5

Doc v1.5

zhangriqi committed:
Merge branch 'master' into PPP

zhangriqi committed:
Merge branch 'master' into 15_April_2

zhangriqi committed:
Merge pull request #198 from Der-Erlkonig/15_April_2

Changes to PPP and DevGuide

zhangriqi committed:
Merge branch 'master' into PPP

melvintzw committed:
Merge pull request #196 from zhangriqi/PPP

Update PPP and documentation

melvintzw committed:
include tags in PPP

melvintzw committed:
Include hyperlinks for examples for bug fix section

zhangriqi committed:
Merge pull request #199 from melvintzw/Doc_v1.5

Doc v1.5

jonleeyz committed:
Merge branch 'master' into documentation

melvintzw committed:
Merge pull request #200 from jonleeyz/documentation

Update: Documentation

jonleeyz committed:
[DeveloperGuide.adoc] Update.

melvintzw committed:
Merge pull request #201 from jonleeyz/documentation-updated

Update: Developer Guide

melvintzw committed:
Change devGuide>implementation>browserPanel image name and width
Collate

jonleeyz committed:
Merge pull request #202 from melvintzw/master

Collate
